### PR TITLE
Replace multiple binds

### LINKED ISSUE #203: Avoid using bind multiple times
I have noticed that there are many places* where bind is used in render method, creating a new function on every invocation. This results in unnecessary memory pollution and re-renders, as a result affecting performance.

A proper approach would be to either:
- Use class property syntax with Babel, or
- Bind every method *once* in constructor

\* few examples:
- https://github.com/allegro/turnilo/blob/master/src/client/views/cube-view/cube-view.tsx#L367
- https://github.com/allegro/turnilo/blob/master/src/client/views/home-view/home-header-bar/home-header-bar.tsx#L68
- https://github.com/allegro/turnilo/blob/master/src/client/components/clearable-input/clearable-input.tsx#L64

### PR DESCRIPTION
**Pull request by pwolaq**
Resolves #203.

It took me some time but I eventually managed to replaced all instances of `.bind(this)` with class property bindings. Additionally I replaced unnecessary constructors in few places with setting state via class property.

This change should not break anything, but I had a lot of conflicts to resolve due to today's merges so please review carefully.

### CODE REVIEW

File: src/client/components/scroller/scroller.tsx [hunk 1]
```diff
@@ -216,7 +216,7 @@ export class Scroller extends React.Component<ScrollerProps, ScrollerState> {
     return ReactDOM.findDOMNode(this.refs[refName]) as any;
   }
 
-  private onScroll = (e: UIEvent) => {
+  private onScroll = (e: React.UIEvent<HTMLElement>) => {
     const { bodyWidth, bodyHeight } = this.props.layout;
     const { viewportWidth, viewportHeight } = this.state;
     var target = e.target as Element;
```
[the remark this answers is not in the document]
Comment (pwolaq) [reply src/client/components/scroller/scroller.tsx]: From what I recall `e.target` was instance of `EventTarget`, which resulted in errors. 

The main purpose of this PR is to get rid of using `bind` so I did not want to spend more time on fixing this (as this is my first experience with Typescript :) ).

[the remark this answers is not in the document]
Comment (adrianmroz) [reply src/client/components/scroller/scroller.tsx]: Sure thing! I'm just writing it down to look into later.

File: src/client/components/clearable-input/clearable-input.tsx [hunk 2]
```diff
@@ -38,34 +38,38 @@ export interface ClearableInputState {
 }
 
 export class ClearableInput extends React.Component<ClearableInputProps, ClearableInputState> {
+  static defaultProps = {
```
Comment (adrianmroz) [inline src/client/components/clearable-input/clearable-input.tsx]: <3

File: src/client/components/date-range-input/date-range-input.tsx [hunk 3]
```diff
@@ -36,14 +36,10 @@ export interface DateRangeInputState {
...
 export class DateRangeInput extends React.Component<DateRangeInputProps, DateRangeInputState> {
-
-  constructor(props: DateRangeInputProps) {
-    super(props);
-    this.state = {
-      dateString: "",
-      timeString: ""
-    };
-  }
+  state = {
```
Comment (adrianmroz) [inline src/client/components/date-range-input/date-range-input.tsx]: Missing type of state. Typescript does not always infer type of state field. I wonder why.

**Pull request merged**